3p/README.md

```diff
@@ -60,3 +60,8 @@ Review the [ads/README](../ads/README.md) for further details on ad integration.
 - Font loading gets controlled (but not initiated) by [`<amp-font>`](https://github.com/ampproject/amphtml/issues/648).
 - AMP by default does not allow inclusion of external stylesheets, but it is happy to whitelist URL prefixes of font providers for font inclusion via link tags. These link tags and their fonts must be served via HTTPS.
 - If a font provider does referrer based "security" it needs to whitelist the AMP proxy origins before being included in the link tag whitelist. AMP proxy sends the appropriate referrer header such as "https://cdn.ampproject.org" and "https://amp.cloudflare.com".
+
+# Adding proper integration tests
+AMP is continuously improving its test coverage and release process to prevent production regression. However, having all types of tests in AMP repo is unrealistic and hard to maintain, especially when the desired test is an end-to-end black box test for a 3rd party integration.
```

Comment:
This reads as though we're discouraging integration tests as part of the AMP repo, but do we really want to do that?

Should we instead say essentially: "You should ensure there are integration tests for your extension.  These should be added to the AMP repo where it makes sense.  In some cases this won't be possible because it relies on bringing up third-party infrastructure.  In these cases you should maintain testing for the extension on your infrastructure against both production AMP and canary."

Reply:
A black box integration normally would involve AMP servers and 3rd party servers. Having these tests in AMP repo could be causing extra flakiness (3P server changes/outage). Meantime, AMP community is not the best one to look at those failures (normally they are business related). In general, I felt it's not scalable in long term to have 3rd party integration tests in AMP.

Reply:
I agree with the sentiment of not adding integration tests that make direct calls to 3rd party services. However, we do currently have non-hermetic integration tests that do hit external resources.

I like Joey's rewording of the message, since it makes it clear that while we want to avoid relying on / bringing up 3rd party infrastructure, extension developers are still encouraged to add tests for their code.

Reply:
SG, updated.